packages/ui/src/AppHeader/SelectCluster/SelectCluster.tsx

```diff
@@ -0,0 +1,81 @@
+import cn from 'classnames'
+import React, { FC, useMemo } from 'react'
+import { CheckCircle } from 'react-feather'
+
+import { Loader } from '../../Loader'
+import { SelectField } from '../../Select'
+
+import styles from './SelectCluster.module.scss'
+
+const getSafeValue = (
+  clusterName: string | null,
+  clusterNames: string[] | null,
+): {
+  status: 'disabled' | 'connected' | 'disconnected'
+  value: string
+} => {
+  if (!clusterNames?.length) {
+    return { status: 'disabled', value: 'No Clusters' }
+  }
+
+  if (!clusterName) {
+    return {
+      status: 'disconnected',
+      value: clusterName || 'Select Cluster',
+    }
+  }
+
+  return {
+    status: clusterNames.includes(clusterName) ? 'connected' : 'disconnected',
+    value: clusterName,
+  }
+}
+
+export type SelectClusterProps = {
+  clusterVersions?: Record<string, string | null> | null
+  clusterName: string | null
+  clusterNames: string[] | null
+  isReadOnly?: boolean
+  onChange: (value: string | null) => void
+}
+
+export const SelectCluster: FC<SelectClusterProps> = ({ clusterVersions, clusterName, clusterNames, isReadOnly, onChange }) => {
+  const options = useMemo(
+    () =>
+      Object.entries(clusterVersions ?? {}).map(([clusterName, version]) => ({
```

Comment:
I guess while we're waiting for `clusterVersions` to be resolved, we'll get no options. Does mapping `clusterNames` here make sense?

Reply:
You tell me ;) I just copied it from MC.

Reply:
It's not critical, so I suggest adding this one to Hive and leaving MC as is, at least until the style update.

Reply:
done